connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfig.java

```diff
@@ -239,6 +257,63 @@ protected static ConfigDef baseConfigDef() {
                         Importance.LOW, HEADER_CONVERTER_CLASS_DOC);
     }
 
+    private void logInternalConverterDeprecationWarnings(Map<String, String> props) {
+        String[] deprecatedConfigs = new String[] {
+            INTERNAL_KEY_CONVERTER_CLASS_CONFIG,
+            INTERNAL_VALUE_CONVERTER_CLASS_CONFIG
+        };
+        for (String config : deprecatedConfigs) {
+            if (props.containsKey(config)) {
+                if (INTERNAL_CONVERTER_DEFAULT.equals(getClass(config))) {
+                    log.info(
+                        "Configuration {} is deprecated and will be removed in an upcoming release. "
+                            + "This value matches the current default, so it can be removed without concern.",
+                        config
+                    );
+                } else {
+                    log.warn(
+                        "Configuration {} is deprecated and will be removed in an upcoming release. "
+                            + "The current value '{}' does NOT match the default of {}, and behavior may change when removed.",
```

Comment:
Okay, now that I read those messages in context, they could be improved since the subject of each sentence is different. The subject of the first sentence "Configuration {} is deprecated and will be removed in an upcoming release" is the configuration definition, whereas in the second sentence "it" refers not to the configuration but to the property set in the worker config file.

How about this:
> Worker configuration property '{}' is deprecated and may be removed in an upcoming release. The specified value '{}' matches the default, so you can safely remove this property from your worker configuration.

and
> Worker configuration property '{}' is deprecated and may be removed in an upcoming release. The specified value '{}' does NOT match the default and recommended value '{}'.